client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -1192,7 +1192,12 @@ class DeltaSharingRestClient(
               val error = s"Request to delta sharing server failed$getDsQueryIdForLogging " +
                 s"due to ${e}."
               logError(error)
-              lineBuffer += error
+              if (lineBuffer.nonEmpty) {
+                val lastIndex = lineBuffer.length - 1
+                lineBuffer(lastIndex) = (error + lineBuffer(lastIndex)).replace(' ', '_')
+              } else {
+                lineBuffer += error.replace(' ', '_')
+              }
```

Comment:
Question: we are returning the entire lineBuffer.toList below; so how will updating last line help? Should we log the lineBuffer?

Reply:
I'm hoping the last line will include the malformed json, because the actual error  as currently it's only the line below. 
We can also try log the last X lines for debugging. 

```Request to delta sharing server failed due to org.apache.http.ConnectionClosedException: Premature end of chunk coded message body: closing chunk expected.```

Reply:
+1 to logging this as well.

Reply:
added